### DISCUSSION TITLE
using setValue does not fix the error

### DISCUSSION THREAD

**Discussion by Amin-Partovi** [react-hook-form/react-hook-form#4768] [Q&A]
Initially, I appreciate the efforts of all the contributors of react-hook-form
I have a custom input which its error is triggered while executing onBlur event. when I update its value using setValue the error persists. when I use {shouldValidate : true} the error disappears as soon as the setValue is executed, but now the problem is that the error exist from the begining of component mounting because the input is initially empty.
my useForm arguments are as follow:
    mode: "all",
    shouldFocusError: true,
    shouldUnregister: false,
I am very grateful for your help

**Comment by bluebill1049** [comment]
provide a reproduceable codesandbox?
